btcec/schnorr/signature_test.go

```diff
@@ -254,3 +256,37 @@ func TestSchnorrVerify(t *testing.T) {
 		}
 	}
 }
+
+// TestSchnorrSignNoMutate tests that generating a schnorr signature doesn't
+// modify/mutate the underlying private key.
+func TestSchnorrSignNoMutate(t *testing.T) {
+	t.Parallel()
+
+	// Assert that given a random private key and message, we can generate
+	// a signature from that w/o modifying the underlying private key.
+	f := func(privBytes, msg [32]byte) bool {
+		privKey, _ := btcec.PrivKeyFromBytes(privBytes[:])
```

Comment:
Since we're passing an array as a slice, `PrivKeyFromBytes()` can mutate `privBytes`.

I think doing a
```
copyPrivBytes:= privBytes
...
// We should be able to re-derive the private key from raw
// bytes and have that match up again.
privKeyCopy, _ := btcec.PrivKeyFromBytes(copyPrivBytes[:])
```

would be more prudent since we now guaranteed that the privKey we're using isn't mutated.

Reply:
I think `copyPrivBytes:= privBytes` still copies the pointer? If we want to copy values we need to declare a new slice `copyPrivBytes := make([]byte, len(privBytes))`, then use `copy`. 

Or we can use array instead of slice.

Reply:
`privBytes` is an array that's being passed as a slice. It's a value not a reference so `:=` will suffice.

I made a runnable example here:
https://go.dev/play/p/B4W-TMo9Xmo

Reply:
oh right, that's already an array. Was thinking maybe `btcec.PrivKeyFromBytes` can take an array, just to be safu.

Reply:
PrivKeyFromBytes doesn't appear to modify the byte slice since it has length 32

Reply:
This change costs almost nothing in terms of performance and guarantees correctness in case of future changes to PrivKeyFromBytes.

No brainer change in my view.

Reply:
Added the copy just to make it a bit more explicit what the test is attempting to assert.